### ISSUE TITLE
Environment variable for less Debug logs? 

### ISSUE THREAD

**Issue by qdm12** [spritsail/plex-media-server#5]
Hi there,

First of all thanks for this great image and maintenance! I see there are a lot of debug lines in my Docker logs, would it be possible to change the verbosity level? An environment variable would be great for that purpose. 

Thanks!

**Comment by frebib** [comment]
It's a configurable option within the Plex web UI by default (you'll have to show advanced).

![](https://i.imgur.com/VhCVUK2h.jpg)

If it appears in Preferences.xml then it can be trivially added to the gen-config script. You're welcome to PR or I can try to do it later.

**Comment by frebib** [comment]
I just had a quick look and it appears there are two fields `logDebug` and `LogVerbose` (nice consistency there Plex), both of which are `0` or `1`. A bit of parsing logic and it should be easy

**Comment by qdm12** [comment]
Okay thanks, I'll try to do a PR today

**Comment by qdm12** [comment]
Hi there ! I am already done with changes to the readme, entrypoint etc. but I cannot build the image. I obtain the error:

```
Step 24/50 : RUN git clone git://git.code.sf.net/p/xmlstar/code --branch $XMLSTAR_VER --depth 1 .  && git apply /tmp/xmlstarlet*.patch  && autoreconf -sif  && ./configure         --prefix=/usr         --disable-build-docs         --with-libxml-prefix=/prefix/usr         --with-libxslt-prefix=/prefix/usr  && make -j$(nproc)  && make -j$(nproc) DESTDIR=/prefix install
 ---> Running in 9c997a44431a
+ git clone git://git.code.sf.net/p/xmlstar/code --branch 1.6.1 --depth 1 .
Cloning into '.'...
Note: checking out '9a470e3db6a632a31d75ed2a6112407e05f92e8d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

+ git apply /tmp/xmlstarlet-0001-Fix-disable-build-docs.patch
/tmp/xmlstarlet-0001-Fix-disable-build-docs.patch:25: trailing whitespace.
man_MANS = $(manpage)
/tmp/xmlstarlet-0001-Fix-disable-build-docs.patch:26: trailing whitespace.
dist_doc_DATA = $(userguide_gen) doc/html.css $(txtguide)
/tmp/xmlstarlet-0001-Fix-disable-build-docs.patch:27: trailing whitespace.
EXTRA_DIST += $(manpage_src) $(userguide_src) $(txtguide_src)
/tmp/xmlstarlet-0001-Fix-disable-build-docs.patch:28: trailing whitespace.
EXTRA_DIST += $(buildfiles_docs) $(manpage)
/tmp/xmlstarlet-0001-Fix-disable-build-docs.patch:29: trailing whitespace.

error: patch failed: Makefile.am:75
error: Makefile.am: patch does not apply
error: patch failed: doc/doc.mk:28
error: doc/doc.mk: patch does not apply
```

It looks related to the xmlstarlet patch, any idea how to solve this? Thanks !

PS: I have put the log modifications in entrypoint as you probably want to change them when rebooting the container, not just at creation in *genconfig.sh*.

**Comment by frebib** [comment]
I'm not sure why it's not building. Are you building on a Windows system? Perhaps try opening a PR and see if the CI can build the patch.

**Comment by qdm12** [comment]
I will try on a Linux system although I really hope Docker works the same everywhere! (it does not, in example /etc/resolv.conf is not the same...)

**Comment by frebib** [comment]
I think the issue could be that the line endings in the patch file have been changed to DOS line endings `\r\n` by git. Either try `dos2unix *` or re-clone the repo with Unix line endings in git

**Comment by qdm12** [comment]
Yes that was it, sorry for the stupid question... I edited the files through Windows and tried to build it on another Debian system... and the line endings, yet again !
I'll open the pull request soon. Thanks !

**Comment by qdm12** [comment]
Hi again, almost done ! Plex works but I have these two errors in the logs when launching Plex (as there is almost no log now so it's easy to spot errors)

```
Nov 21, 2018 13:50:03.230 [0x7fd86deef480] ERROR - Transcoder: Error cleaning old transcode sessions: boost::filesystem::directory_iterator::construct: No such file or directory: "/transcode/Transcode/Sessions"
Nov 21, 2018 13:50:03.268 [0x7fd86deef480] ERROR - Gracenote: Couldn't start up (-1868562431) path to given sqlite3 shared library path valid
```

Any idea what's wrong?

**Comment by frebib** [comment]
Are you mounting `/transcode` inside the container? Make sure that the Plex user has access to write into that directory. Failing that, post your `docker run` line/`docker-compose.yml` and I'll try to reproduce

**Comment by frebib** [comment]
Are you mounting `/transcode` inside the container? Make sure that the Plex user has access to write into that directory. Failing that, post your `docker run` line/`docker-compose.yml` and I'll try to reproduce

**Comment by qdm12** [comment]
Yes indeed, the first error is gone after running the Transcode in persistent storage.

I still have the error `ERROR - Gracenote: Couldn't start up (-1868562431) path to given sqlite3 shared library path valid` although not a deal-break, it works.

I will open the PR now, here is my *docker-compose.yml* more or less

```yml
version: '3'
services:
  plex:
    image: spritsail/plex-media-server
    volumes:
      - ./config:/config
      - ./transcode:/transcode
      - ./data:/data:ro
    network_mode: bridge
    environment:
      - SUID=1000
      - SGID=1000
      - PLEX_CLAIM=claim-blablabla
      - ADVERTISE_IP=https://plex.mydomain.com
      - LOGVERBOSE=0
      - LOGDEBUG=0
```

**Comment by frebib** [comment]
Closed by f9e4de3149fc34dabcfe202ff722ac20fbd00b0a
